trino-s3-proxy/src/main/java/io/trino/s3/proxy/server/rest/TrinoS3ProxyClient.java

```diff
@@ -93,14 +93,19 @@ public void proxyRequest(SigningMetadata signingMetadata, ContainerRequest reque
         MultivaluedMap<String, String> realRequestHeaders = new MultivaluedHashMap<>();
         request.getRequestHeaders().forEach((key, value) -> {
             switch (key.toLowerCase()) {
-                case "x-amz-security-token" -> {}   // we don't use sessions when making the real AWS call
+                case "x-amz-security-token" -> {}  // we add this below
```

Comment:
we can combine this case with the one bellow since they all no-op

Reply:
I like having the comment, though. I'll leave it for now.